egs/babel_multilang/s5/local/nnet3/run_tdnn_joint_babel_sp_bnf.sh

```diff
@@ -0,0 +1,241 @@
+  mkdir -p data/multi
+  mkdir -p data/multi/train${suffix}_hires
+  combine_lang_list=""
+  for lang in `seq 0 $[$num_lang-1]`;do
+    combine_lang_list="$combine_lang_list data/${lang_list[$lang]}/train${suffix}_hires"
+  done
+  utils/combine_data.sh data/multi/train${suffix}_hires $combine_lang_list
+  utils/validate_data_dir.sh --no-feats data/multi/train${suffix}_hires
+  touch data/multi/train${suffix}_hires/.done
+fi
+# If we do not use separate initial layer per language
+# then we use ivector extractor trained on pooled data from all languages
+# using an LDA+MLLT transform arbitrarily chonsed from single language.
+echo hi
+if [ ! -f $global_extractor/.done ]; then
+  echo "Generate global i-vector extractor"
+  local/nnet3/run_shared_ivector_extractor.sh --global-extractor $global_extractor \
+    --stage $stage $lda_mllt_lang || exit 1; 
+  touch $global_extractor/.done
+fi
+
+# extract ivector for all languages.
+for lang in `seq 0 $[$num_lang-1]`; do
+  local/nnet3/run_ivector_common_langs.sh --stage $stage \
+    --global-extractor $global_extractor \
+    --speed-perturb $speed_perturb ${lang_list[$lang]} || exit;
+done
+
+# set num_leaves for all languages
+for lang in `seq 0 $[$num_lang-1]`; do
+  num_leaves=`tree-info exp/${lang_list[$lang]}/$alidir/tree 2>/dev/null | grep num-pdfs | awk '{print $2}'` || exit 1;
+  num_multiple_leaves="$num_multiple_leaves $num_leaves"
+
+  multi_egs_dirs[$lang]=exp/${lang_list[$lang]}/nnet3/egs_v${avg_num_archives}${ivector_suffix}
+  multi_ali_dirs[$lang]=exp/${lang_list[$lang]}/tri5_ali${suffix}
+
+done
+
+online_ivector_dir=
+if $use_ivector; then
+  online_ivector_dir=exp/${lang_list[0]}/nnet3/ivectors_train${suffix}${ivector_suffix}
+fi
+if [ -z "${online_ivector_dir}" ]; then
+  ivector_dim=0
+else
+  ivector_dim=$(feat-to-dim scp:${online_ivector_dir}/ivector_online.scp -) || exit 1;
+fi
+feat_dim=`feat-to-dim scp:data/${lang_list[0]}/train${suffix}${aux_suffix}/feats.scp -`
+
+
+if [ $stage -le 9 ]; then
+  mkdir -p $dir/log
+  echo "$0: creating neural net config for multilingual setups"
+   # create the config files for nnet initialization
+  $cmd $dir/log/make_config.log \
+  python steps/nnet3/multi/make_configs.py  \
+    --splice-indexes "$splice_indexes"  \
+    --feat-dim $feat_dim \
+    --ivector-dim $ivector_dim  \
+    --relu-dim $relu_dim \
+    --use-presoftmax-prior-scale false \
+    --num-multiple-targets  "$num_multiple_leaves"  \
+    --add-lda false \
+    --bottleneck-dim $bottleneck_dim --bottleneck-layer $bnf_layer \
+   $dir/configs || exit 1;
+  # Initialize as "raw" nnet, prior to training the LDA-like preconditioning
+  # matrix.  This first config just does any initial splicing that we do;
+  # we do this as it's a convenient way to get the stats for the 'lda-like'
+  # transform.
+  $cmd $dir/log/nnet_init.log \
+    nnet3-init --srand=-2 $dir/configs/init.config $dir/init.raw || exit 1;
+fi
+
+. $dir/configs/vars || exit 1;
+
+if [ $stage -le 10 ]; then
```

Comment:
Are you sure you want to all this in a top level script in `local/`. I would recommend moving these steps to some script like `steps/nnet3/multilingual/get_egs.sh`